centreon/packages/ui/src/Listing/useStyleTable.ts

```diff
@@ -0,0 +1,83 @@
+import { useEffect } from 'react';
+
+import { useAtomValue, useUpdateAtom } from 'jotai/utils';
+import { equals, isNil, pick } from 'ramda';
+import { CSSObject } from 'tss-react';
+
+import { Theme } from '@mui/material';
+
+import { ListingVariant } from '@centreon/ui-context';
+
+import { Column, TableStyleAtom as Style } from './models';
+import { tableStyleAtom, tableStyleDerivedAtom } from './tableAtoms';
+
+interface TableStyle {
+  checkable?: boolean;
+  currentVisibleColumns?: Array<Column>;
+  viewMode?: ListingVariant;
+}
+
+interface Table {
+  dataStyle: Style;
+  getGridTemplateColumn: string;
+}
+
+const isCompactMode = equals<ListingVariant | undefined>(
+  ListingVariant.compact
+);
+
+interface GetTextStyleProps {
+  theme: Theme;
+  viewMode?: ListingVariant;
+}
+
+export const getTextStyleByViewMode = ({
+  viewMode,
+  theme
+}: GetTextStyleProps): CSSObject =>
+  pick(
+    ['color', 'fontSize', 'lineHeight'],
+    theme.typography[isCompactMode(viewMode) ? 'body2' : 'body1']
+  );
+
+const useStyleTable = ({
+  checkable,
+  currentVisibleColumns,
+  viewMode
+}: TableStyle): Table => {
+  const dataStyle = useAtomValue(tableStyleAtom);
+
+  const updateStyleTable = useUpdateAtom(tableStyleDerivedAtom);
+
+  const getGridTemplateColumn = (): string => {
+    const checkbox = checkable ? '50px ' : '';
+
+    const columnTemplate = currentVisibleColumns
+      ?.map(({ width, shortLabel }) => {
+        if (!isNil(shortLabel)) {
+          return 'min-content';
+        }
+        if (isNil(width)) {
+          return 'auto';
+        }
+
+        return typeof width === 'number' ? `${width}px` : width;
```

Comment:
```suggestion
        return equals(type(width), 'Number') ? `${width}px` : width;
```
Those function are imported from Ramda